go/integration_test/processor/transfer_test.go

```diff
@@ -0,0 +1,218 @@
+				}
+				transaction := tosca.Transaction{
+					Sender:    sender0,
+					Recipient: &receiver0,
+					GasLimit:  sufficientGas,
+				}
+
+				transactionContext := newScenarioContext(state)
+
+				// Run the processor
+				result, err := processor.Run(tosca.BlockParameters{}, transaction, transactionContext)
+				if err != nil || !result.Success {
+					t.Fatalf("execution was not successful or failed with error %v", err)
+				}
+
+				// static call and delegate call do not transfer value,
+				// therefore the transaction does not fail with insufficient balance
+				if call.callType == vm.STATICCALL || call.callType == vm.DELEGATECALL {
+					if !slices.Equal(result.Output, append(bytes.Repeat([]byte{0}, 31), checkValue)) {
+						t.Errorf("transaction has not been handled correctly, code was not executed")
+					}
+				} else {
+					if !slices.Equal(result.Output, bytes.Repeat([]byte{0}, 32)) {
+						t.Errorf("transaction has not been handled correctly, code was executed")
+					}
+				}
+
+				zero := tosca.NewValue(0)
+				if balance := transactionContext.GetBalance(sender0); balance.Cmp(zero) != 0 {
+					t.Errorf("sender balance was not updated correctly, want %v, got %v", zero, balance)
+				}
+				if balance := transactionContext.GetBalance(receiver0); balance.Cmp(receiverBalance) != 0 {
+					t.Errorf("receiver balance was not updated correctly, want %v, got %v", receiverBalance, balance)
+				}
+				if balance := transactionContext.GetBalance(receiver1); balance.Cmp(zero) != 0 {
+					t.Errorf("receiver balance was not updated correctly, want %v, got %v", zero, balance)
+				}
+			})
+		}
+	}
+}
+
+func TestProcessor_TransferToSelf(t *testing.T) {
+	tests := map[string]struct {
+		value   tosca.Value
+		success bool
+	}{
+		"sufficient balance": {
+			tosca.NewValue(100),
+			true,
+		},
+		"insufficient balance": {
+			tosca.NewValue(10000),
+			false,
+		},
+	}
+	for processorName, processor := range getProcessors() {
+		for testName, test := range tests {
+			t.Run(fmt.Sprintf("%s-%s", processorName, testName), func(t *testing.T) {
+				sender := tosca.Address{1}
+				senderBalance := tosca.NewValue(1000)
+
+				state := WorldState{
+					sender: Account{Balance: senderBalance},
+				}
+				transaction := tosca.Transaction{
+					Sender:    sender,
+					Recipient: &sender,
+					GasLimit:  sufficientGas,
+					Value:     test.value,
+				}
+
+				transactionContext := newScenarioContext(state)
+
+				// Run the processor
+				result, err := processor.Run(tosca.BlockParameters{}, transaction, transactionContext)
+				if err != nil {
+					t.Fatalf("execution failed with error %v", err)
+				}
+				if result.Success != test.success {
+					t.Errorf("expected success flag to be %v, got %v", test.success, result.Success)
+				}
+				if balance := transactionContext.GetBalance(sender); balance.Cmp(senderBalance) != 0 {
+					t.Errorf("sender balance was not updated correctly, want %v, got %v", senderBalance, balance)
```

Comment:
nit: also here, there is no expectation that the balance should be updated at all, right?